doc/ADDB_Monitoring.rst

```diff
@@ -0,0 +1,281 @@
+
+ Addition / Deletion of ADDB monitors
+======================================
+
+Monitors can be added and removed dynamically through configuration. But, currently they would be hard-coded in the addb sub-system. 
+
+During addb sub-system initialization all the monitors needs to be added using m0_addb_monitor_add() API & during addb sub-system finalization all the added monitors needs to be deleted/removed using m0_addb_monitor_del() API.
+
+ Invocation of ADDB monitors
+============================= 
+
+When any addb record is posted on a node(whether a client or a server), we invoke all the monitor’s am_watch() API, present in the global monitor list. These monitors do relevant work only for the addb record types that they want to monitor & for the rest they do nothing. During execution of am_watch(), a monitor can create an addb record and also post it. This posting of addb records from inside monitor is to be done using the AST (Asynchronous system trap) mechanism provided by the Motr.  
+
+ Execution of monitor
+======================
+
+Partial execution i.e. top half is executed for all the addb records during monitor invocation. This just updates the monitor specific summary data or posts an ast if an exception had been encountered.
+
+There is a periodic posting of these addb summary records and this is done by the locality thread on the m0d. This thread also runs the ASTs that are posted for the exception summary records. 
+
+The bottom half i.e. AST part would be run by a dedicated thread & would be synchronized among the various others threads that would run monitors with a sm (state machine) group lock.
+
+ Conformance
+===============
+
+- [i.addb.monitor.add] An API is made available for this. 
+
+- [i.addb.monitor.remove] An API is made available for this. 
+
+- [i.addb.monitor.generate-summary-addb-records] Monitor’s am_watch() function will do this. 
+
+- [r.addb.monitor.deliver-addb-record-to-stats-service] Addition to current ADDB mechanism is to be done to differentiate between summary stats records generated by monitors and other addb records & send these summary records to stats service. 
+
+- [r.addb.monitor.deliver-addb-exception-records-to-local-HA] Monitor’s am_watch() function will do this. 
+
+- [r.addb.monitor.deliver-addb-record-to-addb-service] This makes use of current implementation. 
+
+- [r.addb.monitor.nesting] Monitors generate addb records which themselves can be monitored. 
+
+- [r.addb.stats-service.state] Implementation of stats service handles this. 
+
+- [r.addb.stats-service.query] Implementation of stats service handles this. 
+
+- [r.addb.stats-service.single-instance] Implementation of stats service handles this.
+
+ Dependencies
```

Comment:
"Dependencies" has the same level with "Functional Specification" and "Logical Specification".

Reply:
Incorporated.